docker/Dockerfile

```diff
@@ -135,19 +134,32 @@ RUN --mount=type=bind,from=rosdep-depend,source=/autoware/src,target=/autoware/s
     --mixin release compile-commands ccache \
   && du -sh ${CCACHE_DIR} && ccache -s
 
-CMD ["/bin/bash"]
-
-FROM autoware-universe as devel
+FROM autoware-core as devel
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
-# Install development tools and artifacts
+# Install rosdep dependencies for universe packages
+COPY --from=rosdep-depend /rosdep-universe-depend-packages.txt /tmp/rosdep-universe-depend-packages.txt
+# hadolint ignore=SC2002
 RUN --mount=type=ssh \
-  --mount=type=cache,target=/var/cache/apt,sharing=locked \
-  ./setup-dev-env.sh -y --module dev-tools openadkit \
-  && pip uninstall -y ansible ansible-core \
-  && apt-get autoremove -y && rm -rf "$HOME"/.cache
+  apt-get update \
+  && cat /tmp/rosdep-universe-depend-packages.txt | xargs apt-get install -y --no-install-recommends \
+  && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/* "$HOME"/.cache
+
+# Set up runtime environment and artifacts
+COPY --from=rosdep-depend /rosdep-exec-depend-packages.txt /tmp/rosdep-exec-depend-packages.txt
+# hadolint ignore=SC2002
+RUN --mount=type=ssh \
+  apt-get update \
+  && cat /tmp/rosdep-exec-depend-packages.txt | xargs apt-get install -y --no-install-recommends \
+  && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/* "$HOME"/.cache
+
```

Comment:
`rosdep-universe-depend-packages.txt` contains all `depend` tag's dependencies so that it also contains all `exec_depend`  tag's dependencies. I think we don't need these lines.
https://docs.ros.org/en/humble/Tutorials/Intermediate/Rosdep.html#depend

```suggestion
# Set up runtime environment and artifacts
COPY --from=rosdep-depend /rosdep-exec-depend-packages.txt /tmp/rosdep-exec-depend-packages.txt
# hadolint ignore=SC2002
RUN --mount=type=ssh \
  apt-get update \
  && cat /tmp/rosdep-exec-depend-packages.txt | xargs apt-get install -y --no-install-recommends \
  && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/* "$HOME"/.cache

```